notebooks/MAST/JWST/JWST_EngineeringDatabase_Retreival/JWST_EDB_Retrieval.ipynb

```diff
@@ -0,0 +1,560 @@
+    "The file names have the form: \n",
+    "\n",
+    "    `<mnemonic_name>-<t_start>-<t_end>.csv`\n",
+    "    \n",
+    "Use a dictionary comprehension to construct a list of file names; these will be passed to the webservice calling function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fnames = ['-'.join([m, v['t_start'], v['t_end']]) + '.csv' for m,v in mnemonics.items()]\n",
+    "print(fnames)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Call the Webservice"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set the (optional) output folder name prior to the webservice call. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Sub-directory where the data files will be written:\n",
+    "subdir = 'edb_test'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now call the EDB web service. The files containing data will be written to your local storage, in the specified subdirectory. \n",
+    "\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "\n",
+    "<span style=\"color:black\">\n",
+    "    The webservice may take a long time (or timeout), depending upon the quantity of data in the timeseries within the chosen date range. The six-hour time interval in this example returns over 80,000 rows.  \n",
+    "    \n",
+    "</span>\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "status = download_edb_datafiles(token, fnames, folder=subdir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare the Data for Analysis\n",
+    "\n",
+    "Create a list of pandas dataframes from the mnemonics data that were just written to disk. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = [pd.read_csv(subdir+'/'+f) for f in fnames]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Make sure the sizes of the dataframes are equal, and take a look at the first dataframe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "print('Dataframes have the same size? {}'.format(len(df[0]) == len(df[1])))\n",
+    "df[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Visualize the Data Tuple\n",
+    "\n",
+    "Create an x-y plot for analysis. This is easy to do by plotting the Pandas dataframes. It is more interesting to add color to indicate successive moves of the FSM. Begin by loading a numeric and some Bokeh plotting libraries. "
```

Comment:
Pandas is capitalized here but lowercase elsewhere up above, we should be consistent.